sdk/formrecognizer/Azure.AI.FormRecognizer/tests/Infrastructure/DocumentModelCache.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using ServiceVersion = Azure.AI.FormRecognizer.DocumentAnalysis.DocumentAnalysisClientOptions.ServiceVersion;
+
+namespace Azure.AI.FormRecognizer.DocumentAnalysis.Tests
+{
+    internal static class DocumentModelCache
+    {
+        public static Dictionary<ModelKey, DisposableDocumentModel> Models { get; } = new();
+
+        public static async Task DeleteModelsAsync()
+        {
+            foreach (var model in Models.Values)
+            {
+                try
+                {
+                    await model.DeleteAsync();
+                }
+                catch
+                {
+                    // ignore
+                }
+            }
+
+            Models.Clear();
+        }
+
+        public readonly struct ModelKey
```

Comment:
A key used by the `Models` dictionary to indicate whether two models are the same. When attempting to build a model, if we already have a cached model with the same `ModelKey`, the cached one will be returned instead of making a service call.